audonnx/core/model.py

```diff
@@ -347,6 +365,64 @@ def _device_to_providers(
     return providers
 
 
+def _concat(
+        y: typing.Dict[str, np.ndarray],
+        shapes: typing.Sequence[typing.List[int]],
+):
+    r"""Flatten dictionary by concatenating values."""
+
+    y = list(y.values())
+
+    # special case if all shapes are [-1]
+    if all(map([-1].__eq__, shapes)):
+        return np.stack(y)
+
+    axis = _concat_axis(shapes)
+    if axis == -1:
+        raise RuntimeError(
+            f'To concatenate outputs '
+            f'number of dimensions, '
+            f'position of dynamic axis, '
+            f'and all dimensions except the last non-dynamic axis '
+            f'must match. '
+            f'This does not apply to: '
+            f'{shapes}'
+        )
+
+    return np.concatenate(y, axis=axis)
+
+
+def _concat_axis(shapes: typing.Sequence[int]) -> int:
+    r"""Return concat dimension or -1 if not possible."""
+
+    # number of dimensions do not match
+    if not len(set(map(len, shapes))) == 1:
+        return -1
+
+    # dynamic axis in different positions
+    if not len(set(map(_dynamic_axis, shapes))) == 1:
+        return -1
+
+    axis = len(shapes[0]) - (2 if shapes[0][-1] == -1 else 1)
```

Comment:
As it is not that obvious

```suggestion
    # select last non-dynamic axis
    axis = len(shapes[0]) - (2 if shapes[0][-1] == -1 else 1)
```